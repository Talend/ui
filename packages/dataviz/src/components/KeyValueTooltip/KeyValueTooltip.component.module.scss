@use '~@talend/bootstrap-theme/src/theme/guidelines' as *;
@use '~@talend/design-tokens/lib/tokens';

$module: key-value-tooltip;

.#{$module} {
	list-style-type: none;
	white-space: nowrap;
	pointer-events: none;
	margin-bottom: 0; // override dl margin

	&__entry {
		display: flex;
		align-items: center;
	}

	&__key {
		margin-top: 0; // override dt margin
		font-weight: $font-weight-regular;
		line-height: $line-height-base;
		color: tokens.$coral-color-neutral-background;

		&::after {
			content: ':';
		}
	}

	&__value {
		margin-left: 5px;
		color: tokens.$coral-color-accent-text-weak-active;
	}

	&--pattern {
		.#{$module}__value {
			color: tokens.$coral-color-danger-text-weak;
		}
	}
}
