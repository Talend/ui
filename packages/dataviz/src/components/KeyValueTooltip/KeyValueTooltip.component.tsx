import React from 'react';
import classNames from 'classnames';
import { FormatValue } from '@talend/react-components';
import { Tooltip } from '../Tooltip/Tooltip.component';
import styles from './KeyValueTooltip.component.scss';
import { ChartStyle } from '../../types';

export interface KeyValueTooltipProps {
	entries: TooltipEntry[];
	chartStyle?: ChartStyle;
}

function KeyValueTooltip({ entries, chartStyle = ChartStyle.VALUE }: KeyValueTooltipProps) {
	return (
		<Tooltip>
			<dl
				className={classNames(
					styles['key-value-tooltip'],
					styles[`key-value-tooltip--${chartStyle}`],
				)}
			>
				{entries.map(({ key, value }) => (
					<div className={styles['key-value-tooltip__entry']} key={key}>
						<dt className={styles['key-value-tooltip__key']}>{key}</dt>
						<dd>
							<FormatValue className={styles['key-value-tooltip__value']} value={value} />
						</dd>
					</div>
				))}
			</dl>
		</Tooltip>
	);
}

export interface TooltipEntry {
	key: string;
	value: string;
}

export default Object.assign(KeyValueTooltip, {
	ChartStyle,
});
