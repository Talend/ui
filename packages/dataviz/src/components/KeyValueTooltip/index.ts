export { default } from './KeyValueTooltip.component';
export type { KeyValueTooltipProps, TooltipEntry } from './KeyValueTooltip.component';
