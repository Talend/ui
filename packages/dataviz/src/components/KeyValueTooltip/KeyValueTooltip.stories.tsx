import { Meta, Story } from '@storybook/react';
import KeyValueTooltip, { KeyValueTooltipProps } from './KeyValueTooltip.component';

const Template: Story<KeyValueTooltipProps> = args => <KeyValueTooltip {...args} />;

export default {
	title: 'Dataviz/KeyValueTooltip',
	component: KeyValueTooltip,
	decorators: [
		MyStory => (
			<div style={{ width: 200 }}>
				<MyStory />
			</div>
		),
	],
	args: {
		entries: [
			{
				key: 'First line',
				value: '  5   0  ',
			},
			{
				key: 'Second line',
				value: '50',
			},
		],
	},
	parameters: {
		chromatic: {
			// To avoid issues with charts, we'll need to improve this later on
			diffThreshold: 0.6,
		},
	},
} as Meta;

export const Default = Template.bind({});

export const Pattern = Template.bind({});
Pattern.args = {
	chartStyle: KeyValueTooltip.ChartStyle.PATTERN,
};
