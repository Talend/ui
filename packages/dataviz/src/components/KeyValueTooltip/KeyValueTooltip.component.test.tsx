import React from 'react';
import { mount } from 'enzyme';
import KeyValueTooltip from './KeyValueTooltip.component';

describe('TooltipContent', () => {
	it('Should render', () => {
		const component = mount(
			<KeyValueTooltip
				entries={[
					{
						key: 'key1',
						value: 'value1',
					},
				]}
			/>,
		);
		expect(component.html()).toMatchSnapshot();
	});
});
