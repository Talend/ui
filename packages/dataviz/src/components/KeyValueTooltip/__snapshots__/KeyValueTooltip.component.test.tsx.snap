// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipContent Should render 1`] = `
<div
  class="theme-dataviz-tooltip"
>
  <div
    class="theme-dataviz-tooltip__title"
  />
  <dl
    class="theme-key-value-tooltip theme-key-value-tooltip--value"
  >
    <div
      class="theme-key-value-tooltip__entry"
    >
      <dt
        class="theme-key-value-tooltip__key"
      >
        key1
      </dt>
      <dd>
        <span
          class="theme-key-value-tooltip__value"
        >
          value1
        </span>
      </dd>
    </div>
  </dl>
</div>
`;
